tools/src/main/scala/scala/scalanative/interflow/Interflow.scala

```diff
@@ -66,101 +98,142 @@ class Interflow(val config: build.Config)(implicit
         todo.dequeue()
       }
     }
+
+  /** Thread-safely adds a symbol to process. */
   def pushTodo(name: nir.Global.Member): Unit =
     todo.synchronized {
       if (!reached.contains(name)) {
         todo.enqueue(name)
         reached += name
       }
     }
+
+  /** Thread-safely accesses the list of symbols to process. */
   def allTodo(): Seq[nir.Global.Member] =
     todo.synchronized {
-      todo.toSeq
+      todo.toSeq // QUESTION: is this a copy?
     }
 
+  /** Returns `true` iff `name` has been processed. */
   def isDone(name: nir.Global.Member): Boolean =
     done.synchronized {
       done.contains(name)
     }
+
+  /** Sets `value` as the optimized form of `name`. */
+  // QUESTION: This should be renamed.
```

Comment:
The problem with this name is that `done` is just a state. So a method like `setDone(thing, state)` should take a `Boolean` state instead of an arbitrary type.

I think a better name would be `setOptimizedForm`.

Reply:
Ok, I get it now. Sure `optimized` is more expressive here. Maybe `optimizedDefn` / `optimizedDefinition` though, instead of `optimizedForm`?